docs/website/docs/dlt-ecosystem/destinations/qdrant.md

```diff
@@ -0,0 +1,222 @@
+---
+title: Qdrant
+description: Qdrant is a high-performance vector search engine/database that can be used as a destination in dlt.
+keywords: [qdrant, vector database, destination, dlt]
+---
+
+# Qdrant
+
+[Qdrant](https://qdrant.tech/) is an open-source, high-performance vector search engine/database. It deploys as an API service, providing a search for the nearest high-dimensional vectors.
+This destination helps you load data into Qdrant from [dlt resources](../../general-usage/resource.md).
+
+## Setup Guide
+
+1. To use Qdrant as a destination, make sure `dlt` is installed with the `qdrant` extra:
+
+```bash
+pip install dlt[qdrant]
+```
+
+2. Next, configure the destination in the dlt secrets file. The file is located at `~/.dlt/secrets.toml` by default. Add the following section to the secrets file:
+
+```toml
+[destination.qdrant.credentials]
+location = "https://your-qdrant-url"
+api_key = "your-qdrant-api-key"
+```
+
+In this setup guide, we are using the [Qdrant Cloud](https://cloud.qdrant.io/) to get a hosted Qdrant instance and the [FastEmbed](https://github.com/qdrant/fastembed) package that is built into the [Qdrant client library](https://github.com/qdrant/qdrant-client) for generating embeddings.
+
+If no configuration options are provided, the default fallback will be `http://localhost:6333` with no API key.
+
+3. Define the source of the data. For starters, let's load some data from a simple data structure:
+
+```python
+import dlt
+from dlt.destinations.qdrant import qdrant_adapter
+
+movies = [
+    {
+        "title": "Blade Runner",
+        "year": 1982,
+    },
+    {
+        "title": "Ghost in the Shell",
+        "year": 1995,
+    },
+    {
+        "title": "The Matrix",
+        "year": 1999,
+    }
+]
+```
+
+4. Define the pipeline:
+
+```python
+pipeline = dlt.pipeline(
+    pipeline_name="movies",
+    destination="qdrant",
+    dataset_name="MoviesDataset",
+)
+```
+
+5. Run the pipeline:
+
+```python
+info = pipeline.run(
+    qdrant_adapter(
+        movies,
+        embed="title",
+    )
+)
```

Comment:
Another problem with this sample is when I run it I get this error, even on the first run:
```
Traceback (most recent call last):
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 371, in load
    runner.run_pool(load.config, load)
  File "/Users/b/dlthub/dlt/dlt/common/runners/pool_runner.py", line 80, in run_pool
    while _run_func():
  File "/Users/b/dlthub/dlt/dlt/common/runners/pool_runner.py", line 73, in _run_func
    run_metrics = run_f.run(cast(TExecutor, pool))
  File "/Users/b/dlthub/dlt/dlt/load/load.py", line 405, in run
    self.load_single_package(load_id, schema)
  File "/Users/b/dlthub/dlt/dlt/load/load.py", line 325, in load_single_package
    applied_update = self._init_client(job_client, schema, expected_update, load_id, job_client.should_truncate_table_before_load, job_client.should_load_data_to_staging_dataset if isinstance(job_client, WithStagingDataset) else None)
  File "/Users/b/dlthub/dlt/dlt/load/load.py", line 307, in _init_client
    applied_update = self._init_dataset_and_update_schema(job_client, expected_update, tables_with_jobs | dlt_tables, truncate_tables)
  File "/Users/b/dlthub/dlt/dlt/load/load.py", line 296, in _init_dataset_and_update_schema
    job_client.initialize_storage(truncate_tables=truncate_tables)
  File "/Users/b/dlthub/dlt/dlt/destinations/qdrant/qdrant_client.py", line 230, in initialize_storage
    self._create_sentinel_collection()
  File "/Users/b/dlthub/dlt/dlt/destinations/qdrant/qdrant_client.py", line 246, in _create_sentinel_collection
    self._create_collection(full_collection_name=self.sentinel_collection)
  File "/Users/b/dlthub/dlt/dlt/destinations/qdrant/qdrant_client.py", line 186, in _create_collection
    self.db_client.create_collection(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/qdrant_client.py", line 1410, in create_collection
    return self._client.create_collection(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/qdrant_remote.py", line 1800, in create_collection
    result: Optional[bool] = self.http.collections_api.create_collection(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api/collections_api.py", line 1024, in create_collection
    return self._build_for_create_collection(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api/collections_api.py", line 82, in _build_for_create_collection
    return self.api_client.request(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api_client.py", line 74, in request
    return self.send(request, type_)
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api_client.py", line 97, in send
    raise UnexpectedResponse.for_response(response)
qdrant_client.http.exceptions.UnexpectedResponse: Unexpected Response: 400 (Bad Request)
Raw response content:
b'{"status":{"error":"Wrong input: Collection `movies_dataset` already exists!"},"time":0.000015585}'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "qdrant_pipeline.py", line 25, in <module>
    info = pipeline.run(
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 109, in _wrap
    step_info = f(self, *args, **kwargs)
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 143, in _wrap
    return f(self, *args, **kwargs)
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 465, in run
    return self.load(destination, dataset_name, credentials=credentials)
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 109, in _wrap
    step_info = f(self, *args, **kwargs)
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 83, in _wrap
    rv = f(self, *args, **kwargs)
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 69, in _wrap
    return f(self, *args, **kwargs)
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 143, in _wrap
    return f(self, *args, **kwargs)
  File "/Users/b/dlthub/dlt/dlt/pipeline/pipeline.py", line 376, in load
    raise PipelineStepFailed(self, "load", l_ex, self._get_load_info(load)) from l_ex
dlt.pipeline.exceptions.PipelineStepFailed: Pipeline execution failed at stage load with exception:

<class 'qdrant_client.http.exceptions.UnexpectedResponse'>
Unexpected Response: 400 (Bad Request)
Raw response content:
b'{"status":{"error":"Wrong input: Collection `movies_dataset` already exists!"},"time":0.000015585}'
```

Reply:
https://github.com/dlt-hub/dlt/pull/724/commits/269d2d40b5b587ac240acd4dbb8ad37ae9bfa159